x/gov/keeper/deposit_test.go

```diff
@@ -161,6 +161,86 @@ func TestDeposits(t *testing.T) {
 	}
 }
 
+func TestDepositAmount(t *testing.T) {
+	testcases := []struct {
+		name            string
+		deposit         sdk.Coins
+		minDepositRatio string
+		err             string
+	}{
+		{
+			name:            "good amount and denoms",
+			deposit:         sdk.NewCoins(sdk.NewInt64Coin("stake", 10000)),
+			minDepositRatio: "0.001",
+		},
+		{
+			name:            "good amount and denoms but not enough balance for zcoin",
+			deposit:         sdk.NewCoins(sdk.NewInt64Coin("stake", 10000), sdk.NewInt64Coin("zcoin", 1)),
+			minDepositRatio: "0.001",
+			err:             "not enough balance",
+		},
+		{
+			name:            "too small amount",
+			deposit:         sdk.NewCoins(sdk.NewInt64Coin("stake", 10)),
+			minDepositRatio: "0.001",
+			err:             "received 10stake but need at least one of the following: 10000stake,10zcoin: minimum deposit is too small",
+		},
+		{
+			name:            "too small amount with another coin",
+			deposit:         sdk.NewCoins(sdk.NewInt64Coin("zcoin", 1)),
+			minDepositRatio: "0.001",
+			err:             "received 1zcoin but need at least one of the following: 10000stake,10zcoin: minimum deposit is too small",
+		},
+		{
+			name:            "bad denom",
+			deposit:         sdk.NewCoins(sdk.NewInt64Coin("euro", 10000)),
+			minDepositRatio: "0.001",
+			err:             "deposit contains invalid denom(s): [euro]; accepted denom(s): [stake zcoin]: invalid deposit denom",
+		},
+		{
+			name:            "mix containing bad and good denom",
+			deposit:         sdk.NewCoins(sdk.NewInt64Coin("stake", 10000), sdk.NewInt64Coin("euro", 10000)),
+			minDepositRatio: "0.001",
+			err:             "deposit contains invalid denom(s): [euro stake]; accepted denom(s): [stake zcoin]: invalid deposit denom",
+		},
+		{
+			name:            "minDepositRatio is zero",
+			deposit:         sdk.NewCoins(sdk.NewInt64Coin("stake", 10)),
+			minDepositRatio: "0.0",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			govKeeper, mocks, _, ctx := setupGovKeeper(t)
+			authKeeper, bankKeeper, stakingKeeper := mocks.acctKeeper, mocks.bankKeeper, mocks.stakingKeeper
+			trackMockBalances(bankKeeper)
+
+			testAddrs := simtestutil.AddTestAddrsIncremental(bankKeeper, stakingKeeper, ctx, 2, sdkmath.NewInt(1000000000000000))
+			authKeeper.EXPECT().AddressCodec().Return(address.NewBech32Codec("cosmos")).AnyTimes()
+
+			params, _ := govKeeper.Params.Get(ctx)
+			params.MinDepositRatio = tc.minDepositRatio
+			params.MinDeposit = append(params.MinDeposit, sdk.NewCoin("zcoin", sdkmath.NewInt(10000))) // coins must be sorted by denom
```

Comment:
can we use `Coins.Add` instead of appending ? it'll sort the coins after addition.